app/factories/bulkrax/object_factory.rb

```diff
@@ -182,12 +182,20 @@ def update_file_set(attrs)
       actor.update_metadata(file_set_attrs)
     end
 
+    def pass_collection_to_be_persisted(parent:, child:)
+      if parent.is_a? Array
+        parent.each { |par| persist_collection_memberships(par, child) }
+      else
+        persist_collection_memberships(parent, child)
+      end
+    end
+
     # Add child to parent's #member_collections
     # Add parent to child's #member_of_collections
-    def persist_collection_memberships(parent:, child:)
+    def persist_collection_memberships(parent, child)
       parent.reject!(&:blank?) if parent.respond_to?(:reject!)
       child.reject!(&:blank?) if child.respond_to?(:reject!)
-      return if parent.blank? || child.blank?
+      return if parent.invalid? || child.invalid?
```

Comment:
this method is only called when a collection has a parent collection listed. the child at this time is a new instance of a collection that hasn't been saved so the next line fails in [active fedora](https://github.com/samvera/active_fedora/blob/v12.1.1/lib/active_fedora/relation/finder_methods.rb#L182) for having no id. (see below)

changing the method to `invalid?` means the method is exited before failing at the "::Hyrax::Collections::NestedCollectionPersistenceService..." line below.

the new collection is then properly saved and the relationship to its parent collection is formed, but idk if this is the best way to go about it.

@orangewolf @sephirothkod @bkiahstroud thoughts?
![Screen Shot 2022-04-21 at 5 56 33 PM](https://user-images.githubusercontent.com/29032869/164756437-f9fb559b-5893-4250-ae80-da277c58542e.png)

Reply:
Hmm... this confuses me for a few reasons: 

1. This was working as is in v2.3.0 when I tested it 
2. > the new collection is then properly saved and the relationship to its parent collection is formed

  2a. The fact that the relationship still gets created properly despite bailing out of this method is odd because, as far as I was aware, this was the only thing that actually created the relationship between the collections, so I'm not sure what's going on there. 

Ultimately, I don't think we should just no-op out of this method 100% of the time. If it requires the objects to be persisted, maybe we could call it after the `object.save!` lines in the `#create_collection` and `#update_collection` methods?

Reply:
thanks kiah. I considered calling the method after the objects are saved too if it's necessary that it be called despite the relationships forming without it.

Reply:
After looking through I agree with moving this after the object.save call. If that causes issues for some reason, we can move it back and do a check on the child collection and save if it hasn't been saved yet.

Reply:
changed the code back to using `.blank?` and moved persisting the collection to after the new collection is saved. this still works successfully.